### ISSUE TITLE
Update data condense commands to support new data organization

### ISSUE THREAD

**Issue by mvogelgesang** [GSA/EDX#802]
## Summary

With the introduction of #756, the schema of the results file has changed. As a result, the `data condense` commands need to support pre and post v1.0 versions of the schema.

Currently, the `collections.ts` file follows a simple structure of collection types as keys and an array of CSV header pairs as values. For example:

```ts
export const collections: CollectionType = {
  default: [
    { id: 'domain', title: 'Domain' },
    { id: 'url', title: 'URL' },
  ],
  gearScans: [
    { id: 'domain', title: 'Domain' },
    { id: 'screenCapture.data[0].imgPath', title: 'image' },
    { id: 'screenCapture.data[0].device', title: 'device' },
    { id: 'screenCapture.data[1].imgPath', title: 'image' },
    { id: 'screenCapture.data[1].device', title: 'device' },
  ],
}
```

The new structure should be able to denote the version number(s) it supports. The system should be able to then determine the corresponding collection given the json file version.

## Acceptance Criteria

- [x] Given a series of folders containing website scan reports from pre and post v1.0.0 release, running the `data condense` command should properly parse each file.
- [x] Update Screenshot facet per comment https://github.com/GSA/EDX/issues/802#issuecomment-1498299485
- [x] Complete mappings for pre and post v1.0.0 schemas
- [x] Write tests to validate collections to ensure id's and titles are in sync between versions. The jsonPath is expected to vary but the other elements need to stay in lockstep.

**Comment by mvogelgesang** [comment]
This is doable by employing the [compare-versions](https://github.com/omichelsen/compare-versions) package. Using compare-versions, we can set rules according to the `scanVersion` found in each json file. The syntax is easy to follow and we have the ability to update the collections individually should the need arise.

```ts
export const collections: CollectionType = {
  uswds: {
    '<1.0.0': [
      { id: 'domain', title: 'Domain' },
      { id: 'scanVersion', title: 'Scan Version' },
      { id: 'uswdsComponents.accordion', title: 'accordion' },
      { id: 'uswdsComponents.alert', title: 'alert' },
      { id: 'uswdsComponents.banner', title: 'banner' },
    ],
    '>1.0.0': [
      { id: 'domain', title: 'Domain' },
      { id: 'scanVersion', title: 'Scan Version' },
      { id: 'uswdsComponents.data.accordion', title: 'accordion' },
      { id: 'uswdsComponents.data.alert', title: 'alert' },
      { id: 'uswdsComponents.data.banner', title: 'banner' },
    ],
  },
}
```

**Comment by mvogelgesang** [comment]
Screenshots and SearchEngine facets present a challenge as the data is placed into an array which makes it harder to access specific data points. Currently, the jsonPath is coded as `screenCapture.data[0].device` or `screenCapture.data[1].device`. This isn't scalable. Instead, a reworking of the structure is required which would allow for direct references to the exact data that is desired.

**Proposed**

```json
{
  "screenshot": {
    "data": {
      "webpage": {
        "desktop": {
          "domain": "www.gsa.gov",
          "url": "https://www.gsa.gov/",
          "imgPath": "data/scans/20230323/www.gsa.gov_desktop_9c27f0afb30a0d9b6e3f5f6715b8fcd1.png"
        },
        "mobile": {
          "domain": "www.gsa.gov",
          "url": "https://www.gsa.gov/",
          "imgPath": "data/scans/20230323/www.gsa.gov_mobile_9c27f0afb30a0d9b6e3f5f6715b8fcd1.png"
        }
      },
      "searchEngine": {
        "google": {
          "mobile": {
            "domain": "google.com",
            "url": "https://google.com/search?q=www.gsa.gov",
            "imgPath": "data/scans/20230323/google.com_mobile_141af95349c949dc08317c4cf82948f6.png"
          },
          "desktop": {
            "domain": "google.com",
            "url": "https://google.com/search?q=www.gsa.gov",
            "imgPath": "data/scans/20230323/google.com_desktop_bf09ffeeb54e6c3bb612933520c59429.png"
          }
        }
      }
    },
    "errors": [],
    "description": "Holds screenshots taken throughout scan including homepage, search engine, and others."
  }
}
```

**Comment by mvogelgesang** [comment]
Tests for the mappings are needed but will require additional work to refactor the `data condense` operation. https://github.com/GSA/EDX/issues/803 has been created to track that work.

**Comment by mvogelgesang** [comment]
Merged into #769 feature branch
